src/main/java/io/vertx/ext/asyncsql/impl/AsyncSQLRowStream.java

```diff
@@ -87,11 +88,17 @@ private void nextRow() {
         // mark as ended if the handler was registered too late
         ended.set(true);
         // automatically close resources
-        close(c -> {
-          if (endHandler != null) {
-            endHandler.handle(null);
-          }
-        });
+        if (rsClosedHandler != null) {
+          // only notify (since the rs is closed by the underlying driver)
+          rsClosedHandler.handle(null);
+        } else {
```

Comment:
https://github.com/vert-x3/vertx-sql-common/blob/master/src/main/asciidoc/java/index.adoc#streaming

As I wrote it is present to preserve the same semantics from jdbc-client where you can receive multiple result sets. When you expect more than 1 result set the a ResultSetClosedHandler is registered and instead of notifying that the stream ended, we notify that the current resultset ended.

No matter what these 2 handlers in the async client are NO-OPs since the async driver always loads the bulk data into RAM and the events are just there to allow say for example a user using the jdbc-client to migrate to the async driver without the need to update their code.